bindings/java/nz/mega/sdk/MegaApiJava.java

```diff
@@ -9357,4 +9362,105 @@ public void getUserEmail(long handle, nz.mega.sdk.MegaRequestListenerInterface l
         megaApi.cancelCreateAccount(createDelegateRequestListener(listener));
     }
+
+    /**
+     * Gets the translated string of an error received in a request.
+     *
+     * @param error MegaError received in the request
+     * @return The translated string
+     */
+    public static String getTranslatedErrorString(MegaError error) {
+        MegaApplication app = MegaApplication.getInstance();
+        if (app == null) {
+            return error.getErrorString();
+        }
+
+        if (error.getErrorCode() > 0) {
+            return app.getString(R.string.api_error_http);
+        }
+
+        switch (error.getErrorCode()) {
+            case API_OK:
+                return app.getString(R.string.api_ok);
+            case API_EINTERNAL:
+                return app.getString(R.string.api_einternal);
+            case API_EARGS:
+                return app.getString(R.string.api_eargs);
+            case API_EAGAIN:
+                return app.getString(R.string.api_eagain);
+            case API_ERATELIMIT:
+                return app.getString(R.string.api_eratelimit);
+            case API_EFAILED:
+                return app.getString(R.string.api_efailed);
+            case API_ETOOMANY:
+                if (error.getErrorString().equals("Terms of Service breached")) {
+                    return app.getString(R.string.api_etoomany_ec_download);
+                } else if (error.getErrorString().equals("Too many concurrent connections or transfers")){
+                    return app.getString(R.string.api_etoomay);
+                } else {
+                    return error.getErrorString();
+                }
+            case API_ERANGE:
+                return app.getString(R.string.api_erange);
+            case API_EEXPIRED:
+                return app.getString(R.string.api_eexpired);
+            case API_ENOENT:
+                return app.getString(R.string.api_enoent);
+            case API_ECIRCULAR:
+                if (error.getErrorString().equals("Upload produces recursivity")) {
+                    return app.getString(R.string.api_ecircular_ec_upload);
+                } else if (error.getErrorString().equals("Circular linkage detected")){
+                    return app.getString(R.string.api_ecircular);
+                } else {
+                    return error.getErrorString();
+                }
+            case API_EACCESS:
+                return app.getString(R.string.api_eaccess);
+            case API_EEXIST:
+                return app.getString(R.string.api_eexist);
+            case API_EINCOMPLETE:
+                return app.getString(R.string.api_eincomplete);
+            case API_EKEY:
+                return app.getString(R.string.api_ekey);
+            case API_ESID:
+                return app.getString(R.string.api_esid);
+            case API_EBLOCKED:
+                return app.getString(R.string.api_eblocked);
```

Comment:
You have forgotten one case:
```suggestion
                if (error.getErrorString().equals("Not accessible due to ToS/AUP violation")) {
                    return app.getString(R.string.api_eblocked_ec_import-download);
                } else {
                    return app.getString(R.string.api_eblocked);
                }
```